packages/fxa-settings/src/lib/glean/index.ts

```diff
@@ -126,8 +126,15 @@ const populateMetrics = async (gleanPingMetrics: GleanPingMetrics) => {
   if (gleanPingMetrics?.event) {
     // The event here is the Glean `event` metric type, not an "metrics event" in
     // a more general sense
+
     for (const n of eventPropertyNames as PropertyNameT) {
-      event[n].set(gleanPingMetrics.event[n] || '');
+      const eventValue = gleanPingMetrics.event[n];
+
+      if (typeof eventValue === 'boolean' && n === 'thirdPartyLinks') {
```

Comment:
I asked Barry for an early review of this [here](https://mozilla.slack.com/archives/CLV3KMZ8B/p1721764215322759) so I know adding the extra event property seems right, but these lines here are not great. I'm happy to circle back on this but LMK if you see what Typescript edits I can make here to improve this.

I also don't want to give a boolean type a boolean default, I'd rather leave it undefined in that case, which is why I've got the changes further down in this file.